File: 234. Palindrome Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 *
 * //转换成数组再进行头尾比较
 *
 */


bool isPalindrome(struct ListNode* head){
    int arr[70000],len=0;
    struct ListNode* pointer=head;
    while(pointer){
        arr[len++]=pointer->val;
        pointer=pointer->next;
    }
    for(int i=0;i<len/2;i++){
        if(arr[i]!=arr[len-i-1]) return false;
    }
    return true;
}

File: 234. Palindrome Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    // 法一
    let cur = head, arr = []
    while(cur) {
        arr.push(cur.val)
        cur = cur.next
    }
    for(let i = 0, j = arr.length - 1; i <= j; i++, j--) {
        if(arr[i] !== arr[j]) return false
    }
    return true

    // todo 链表
};